File: .gemini/antigravity/brain/a05409aa-156b-4713-8f70-2d4225d9cd16/scratch/clean_imports.py
import os
import re

def clean_imports(directory):
    # Regex to match the unnecessary imports
    # We want to match both single and double quotes
    patterns = [
        re.compile(r"^import\s+['\"]package:flutter_animate/flutter_animate.dart['\"];\s*$"),
        re.compile(r"^import\s+['\"]package:font_awesome_flutter/font_awesome_flutter.dart['\"];\s*$")
    ]

    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith('.dart'):
                file_path = os.path.join(root, file)
                
                # Skip the barrel file itself if it's in this directory
                if file == 'imports.dart':
                    continue
                    
                with open(file_path, 'r', encoding='utf-8') as f:
                    lines = f.readlines()
                
                new_lines = []
                changed = False
                for line in lines:
                    is_redundant = False
                    for p in patterns:
                        if p.match(line.strip()):
                            is_redundant = True
                            break
                    
                    if is_redundant:
                        changed = True
                    else:
                        new_lines.append(line)
                
                if changed:
                    with open(file_path, 'w', encoding='utf-8') as f:
                        f.writelines(new_lines)
                    print(f"Cleaned {file_path}")

if __name__ == "__main__":
    lib_dir = os.path.join(os.getcwd(), 'lib')
    if os.path.exists(lib_dir):
        clean_imports(lib_dir)
    else:
        print("lib directory not found")

File: .gemini/antigravity/brain/a05409aa-156b-4713-8f70-2d4225d9cd16/scratch/clean_imports.dart
import 'dart:io';

void main() {
  final libDir = Directory('lib');
  if (!libDir.existsSync()) {
    print('lib directory not found');
    return;
  }

  final redundant = {
    "import 'package:flutter_animate/flutter_animate.dart';",
    'import "package:flutter_animate/flutter_animate.dart";',
    "import 'package:font_awesome_flutter/font_awesome_flutter.dart';",
    'import "package:font_awesome_flutter/font_awesome_flutter.dart";',
  };

  libDir.listSync(recursive: true).forEach((entity) {
    if (entity is File && entity.path.endsWith('.dart')) {
      if (entity.path.endsWith('imports.dart')) return;

      final lines = entity.readAsLinesSync();
      final newLines = <String>[];
      bool changed = false;

      for (final line in lines) {
        if (redundant.contains(line.trim())) {
          changed = true;
          continue;
        }
        newLines.add(line);
      }

      if (changed) {
        entity.writeAsStringSync(newLines.join('\n') + '\n');
        print('Cleaned ${entity.path}');
      }
    }
  });
}
